crates/control/src/services/connectors.rs

```diff
@@ -1,17 +1,86 @@
 use std::path::Path;
 
+use anyhow::anyhow;
+use models::Object;
 use tokio::process::Command;
 
 use crate::config::settings;
+use crate::controllers::json_api::RawJson;
+use crate::models::connector_images::ConnectorImage;
+use crate::models::connectors::{Connector, ConnectorOperation};
+use crate::models::names::CatalogName;
+use crate::services::discovery::{DiscoverResponse, DiscoveredCatalog};
+use crate::services::subprocess::Subprocess;
 
-pub fn spec(image: &str) -> Command {
+#[derive(Debug, thiserror::Error)]
+pub enum ConnectorError {
+    #[error("could not execute the connector")]
+    ConnectorFailed(anyhow::Error),
+    #[error("failed to parse configuration")]
+    MalformedConfig(serde_json::Error),
+    #[error("could not parse the connector output")]
+    MalformedOutput(serde_json::Error),
+    #[error("connector does not support operation: `{0}`")]
+    UnsupportedOperation(ConnectorOperation),
+}
+
+pub async fn spec(image: &ConnectorImage) -> Result<RawJson, ConnectorError> {
```

Comment:
What's the rationale for using `RawJson` instead of a typed struct here? Is that just temporary scaffolding, or is the idea to just try to use `RawJson` where we can and let `flowctl api` control the format?